include/unifex/when_all_range.hpp

```diff
@@ -0,0 +1,341 @@
+  void set_value(Value&&... value) noexcept {
+    UNIFEX_TRY {
+      op_.holders_[index_].value.emplace(std::forward<Value>(value)...);
+      op_.element_complete();
+    }
+    UNIFEX_CATCH(...) {
+      this->set_error(std::current_exception());
+    }
+  }
+
+  template <typename Error>
+  void set_error(Error&& error) noexcept {
+    if (!op_.doneOrError_.exchange(true, std::memory_order_relaxed)) {
+      op_.error_.emplace(
+          unifex::in_place_type_t<std::decay_t<Error>>{}, (Error &&) error);
+      op_.stopSource_.request_stop();
+    }
+    op_.element_complete();
+  }
+
+  void set_done() noexcept {
+    if (!op_.doneOrError_.exchange(true, std::memory_order_relaxed)) {
+      op_.stopSource_.request_stop();
+    }
+    op_.element_complete();
+  }
+
+  [[nodiscard]] Receiver& get_receiver() const noexcept {
+    return op_.receiver_;
+  }
+
+  template(typename CPO, typename R)                         //
+      (requires unifex::is_receiver_query_cpo_v<CPO> AND     //
+           unifex::same_as<R, type> AND                      //
+               unifex::is_callable_v<CPO, const Receiver&>)  //
+      friend auto tag_invoke(CPO cpo, const R& r)            //
+      noexcept(unifex::is_nothrow_callable_v<CPO, const Receiver&>)
+          -> unifex::callable_result_t<CPO, const Receiver&> {
+    return std::move(cpo)(std::as_const(r.get_receiver()));
+  }
+
+  [[nodiscard]] unifex::inplace_stop_source& get_stop_source() const noexcept {
+    return op_.stopSource_;
+  }
+
+  friend unifex::inplace_stop_token tag_invoke(
+      unifex::tag_t<unifex::get_stop_token> /* unused */,
+      const type& r) noexcept {
+    return r.get_stop_source().get_token();
+  }
+
+  template <typename Func>
+  friend void tag_invoke(
+      unifex::tag_t<unifex::visit_continuations> /* unused */,
+      const type& r,
+      Func&& func) noexcept {
+    std::invoke(func, r.get_receiver());
+  }
+};
+
+// Sender adapter for vector<Sender>
+template <typename Sender>
+struct _sender {
+  struct type;
+};
+
+template <typename Sender>
+using sender = typename _sender<Sender>::type;
+
+template <typename Sender>
+struct _sender<Sender>::type final {
+public:
+  using sender_value_type =
+      unifex::sender_single_value_result_t<unifex::remove_cvref_t<Sender>>;
+
+  explicit type(std::vector<Sender>&& senders) noexcept(
+      std::is_nothrow_move_constructible_v<std::vector<Sender>>)
+    : senders_(std::move(senders)) {}
+
+  template <
+      template <typename...>
+      class Variant,
+      template <typename...>
+      class Tuple>
+  using value_types = Variant<Tuple<std::vector<sender_value_type>>>;
+
+  template <template <typename...> class Variant>
+  using error_types = Variant<std::exception_ptr>;
+
+  static constexpr bool sends_done = true;
+
+private:
+  template(typename Sender2, typename Receiver)                          //
+      (requires unifex::same_as<unifex::remove_cvref_t<Sender2>, type>)  //
```

Comment:
We also require that `Receiver` is a _Receiver_.  I think this suggestion captures that.
```suggestion
      (requires unifex::same_as<unifex::remove_cvref_t<Sender2>, type> AND //
          unifex::receiver<Receiver>)  //
```